### PR TITLE
347

### PR DESCRIPTION
**Pull request by superspartak**

### CODE REVIEW

File: src/components/molecules/TaskCandidatesSection/TaskCandidatesSection.tsx [hunk 1]
```diff
@@ -67,15 +74,18 @@ const TaskCandidatesSection: FC<TaskCandidatesSectionProps> = ({
...
       }),
     [candidates, getCandidateStatus]
   )
 
-  const handleDelete = useCallback((candidateId: string) => {
-    // TODO: remove candidate with id: candidateId from this task
-  }, [])
+  const handleDelete = useCallback(
+    (vendor_id: string) => {
+      deleteAssignmentVendor({ data: [{ vendor_id }] })
```
Comment (VaabelMaret) [inline src/components/molecules/TaskCandidatesSection/TaskCandidatesSection.tsx]: [blocker]
Deleting error/success message is missing

File: src/components/organisms/tables/SelectVendorsTable/SelectVendorsTable.tsx [hunk 2]
```diff
@@ -88,14 +88,19 @@ const SelectVendorsTable = <TFormValues extends FieldValues>({
   handleSortingChange,
   handlePaginationChange,
```
Comment (VaabelMaret) [inline src/components/organisms/tables/SelectVendorsTable/SelectVendorsTable.tsx:89]: [blocker]

filtering and sorting not working.
Skills are missing, but filtering works.

Comment (VaabelMaret) [reply src/components/organisms/tables/SelectVendorsTable/SelectVendorsTable.tsx:89]: [blocker]

In 'FreeDaysHearder' if we don't have info text then info icon should be hidden

**Review by VaabelMaret — CHANGES_REQUESTED** [review]

**Comment by IngaSokman** [comment]
[blocker] (maybe?) Noticed that when I have already selected a vendor from the list, and then went back to adding vendor modal, cleared filters, then the person that I previously selected, the checkbox is not ticked for that vendor

**Comment by IngaSokman** [comment]
[blocker] On the modal, when I press "Kuvan korraga" button, nothing happens

**Comment by IngaSokman** [comment]
[non-blocker] Getting eslint error in useUsers file, just format it :D

File: src/components/molecules/TaskCandidatesSection/TaskCandidatesSection.tsx (hunk 1 above)
Comment (superspartak) [reply src/components/molecules/TaskCandidatesSection/TaskCandidatesSection.tsx]: Do we use showNotification for that? Is there a specific message described anywhere?

Comment (VaabelMaret) [reply src/components/molecules/TaskCandidatesSection/TaskCandidatesSection.tsx]: yes, we use showNotification. we don't have the messages, but make them all similar.

**Comment by superspartak** [comment]
> [blocker] On the modal, when I press "Kuvan korraga" button, nothing happens

BE related, being fixed

**Comment by superspartak** [comment]
> [blocker] (maybe?) Noticed that when I have already selected a vendor from the list, and then went back to adding vendor modal, cleared filters, then the person that I previously selected, the checkbox is not ticked for that vendor

Good point. What I understood from previous discussions is that this modal is used only to add vendors. If the added vendors are pre-selected, it would give the impression that you can remove them by un-selecting the vendor imo. But the modal is only for adding. If the same vendor is selected again and were tried to be added, it shows the user an error about that. So I would say it works as intended.

File: src/components/molecules/TaskCandidatesSection/TaskCandidatesSection.tsx (hunk 1 above)
Comment (superspartak) [reply src/components/molecules/TaskCandidatesSection/TaskCandidatesSection.tsx]: Done, please check

File: src/components/organisms/tables/SelectVendorsTable/SelectVendorsTable.tsx (hunk 2 above)
Comment (superspartak) [reply src/components/organisms/tables/SelectVendorsTable/SelectVendorsTable.tsx:89]: FreeDaysHearder proper information added, thx to Birgit.

Filtering on languages now works. Skills are added. When you say sorting not working, what do you mean? Which column does not work?

**Review by IngaSokman — APPROVED** [review]

**Pull request merged**